docs/basics/101-126-config2.rst

```diff
@@ -0,0 +1,285 @@
+
+   $ cat recordings/longnow/.gitattributes
+
+The relevant part is ``README.md annex.largefiles=nothing``.
+This instructs Git-annex to specifically not annex ``README.md``.
+
+Lastly, if you wanted to configure a rule based on **size**, you could add a row such as::
+
+   ** annex.largefiles(largerthan=20kb)
+
+to store only files exceeding 20KB in size in Git-annex [#f2]_.
+
+As you may have noticed, unlike ``.git/config`` files,
+there can be multiple ``.gitattributes`` files within a dataset. So far, you have seen one
+in the root of the superdataset, and in the root of the ``longnow`` subdataset.
+In principle, you can add one to every level of your dataset. For example, there is another
+``.gitattributes`` file within the ``.datalad`` directory:
+
+.. runrecord:: _examples/DL-101-125-107
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ cat .datalad/.gitattributes
+
+As with Git configuration files, more specific or lower-level configurations take precedence
+over more general or higher-level configurations. Specifications in a subdirectory can
+therefore overrule specifications made in the ``.gitattributes`` file of the parent
+directory.
+
+In summary, the ``.gitattributes`` files will give you the possibility to configure
+what should be annexed and what shouldn't be annexed up to individual file level.
+This can be very handy, and allows you to tune your dataset to your custom needs.
+For example, files you will often edit by hand could be stored in Git if they are
+not too large to ease modifying them [#f3]_.
+Once you know the basics of this type of configuration syntax, writing
+your own rules is easy. For more tips on how configure Git-annex's content
+management in ``.gitattributes``, take a look at `this <https://git-annex.branchable.com/tips/largefiles/>`_
+page of the Git-annex documentation.
+Later however you will see preconfigured DataLad *procedures* such as ``text2git`` that
+can apply useful configurations for you, just as ``text2git`` added the last line
+in the root ``.gitattributes`` file.
+
+``.gitmodules``
+^^^^^^^^^^^^^^^
+
+On last configuration file we want to look at is the ``.gitmodules`` file.
+There is one right in the root of your dataset:
+
+.. runrecord:: _examples/DL-101-125-108
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ cat .gitmodules
+
+Based on these contents, you might have already guessed what this file
+stores. ``.gitmodules`` is a configuration file that stores the mapping between
+your own dataset and any subdatasets you have installed in it.
+There will be an entry for each submodule (subdataset) in your dataset.
+The name *submodule* is Git terminology, and describes a Git repository inside of
+another Git repository, i.e. the super- and subdataset principles.
+Upon sharing your dataset, the information about subdatasets and where to retrieve
+them from is stored and shared with this file.
+
+
+
+``.datalad/config``
+^^^^^^^^^^^^^^^^^^^
```

Comment:
Worth mentioning that, because this config file comes with any dataset, a local configuration should actually override whatever config comes with a dataset. Otherwise, a git pull could suddenly and unintentionally alter local datalad behavior that was specifically configured.